django/contrib/auth/forms.py

```diff
@@ -189,17 +192,26 @@ def clean(self):
         password = self.cleaned_data.get('password')
 
         if username is not None and password:
-            self.user_cache = authenticate(self.request, username=username, password=password)
-            if self.user_cache is None:
-                raise forms.ValidationError(
-                    self.error_messages['invalid_login'],
-                    code='invalid_login',
-                    params={'username': self.username_field.verbose_name},
+            if len(username) <= self.username_field.max_length:
```

Comment:
It looks incorrect to reimplement a field's validation in `clean()`. Why isn't the field validating the max_length?